Build of http client fails if CONFIG_POSIX_API=y

**Describe the bug**
It is not possible to use CONFIG_POSIX_API together with the http client library.

**To Reproduce**
Steps to reproduce the behavior:
1. Activate for instance for samples/net/sockets/http_client CONFIG_POSIX_API
2. Build the example for a non-posix board, e.g.: west build -p auto -d build samples/net/sockets/http_client/ --board=stm32h735g_disco
3. zephyr/samples/net/sockets/http_client/src/main.c:110: undefined reference to `send'

**Expected behavior**
It should be possible to use the http client library together with CONFIG_POSIX_API

**Impact**
Hinders me from using the already existing http-library in zephyr as I need CONFIG_POSIX_API for a different reason.

--- a/subsys/net/lib/http/http_client.c
+++ b/subsys/net/lib/http/http_client.c
@@ -32,7 +32,7 @@ LOG_MODULE_REGISTER(net_http, CONFIG_NET_HTTP_LOG_LEVEL);
 static ssize_t sendall(int sock, const void *buf, size_t len)
 {
 	while (len) {
-		ssize_t out_len = send(sock, buf, len, 0);
+		ssize_t out_len = zsock_send(sock, buf, len, 0);
 
 		if (out_len < 0) {
 			return -errno;
@@ -415,7 +415,7 @@ static int http_wait_data(int sock, struct http_request *req)
 	int received, ret;
 
 	do {
-		received = recv(sock, req->internal.response.recv_buf + offset,
+		received = zsock_recv(sock, req->internal.response.recv_buf + offset,
 				req->internal.response.recv_buf_len - offset,
 				0);
 		if (received == 0) {
@@ -460,7 +460,7 @@ static void http_timeout(struct k_work *work)
 	struct http_client_internal_data *data =
 		CONTAINER_OF(work, struct http_client_internal_data, work);
 
-	(void)close(data->sock);
+	(void)zsock_close(data->sock);
 }
 
 int http_client_req(int sock, struct http_request *req,

